code/modules/spells/roguetown/acolyte/xylix.dm

```diff
@@ -35,26 +35,25 @@
 	antimagic_allowed = TRUE
 	recharge_time = 30 SECONDS
 	var/firstcast = TRUE
+	var/icon/clone_icon
 
 /obj/effect/proc_holder/spell/invoked/mastersillusion/cast(list/targets, mob/living/carbon/human/user = usr)
-	var/icon/I 
 	if(firstcast)
 		to_chat(user, span_italics("...Oh, oh, thy visage is so grand! Let us prepare it for tricks!"))
-		I = get_flat_human_icon("[user.real_name] decoy", null, null, DUMMY_HUMAN_SLOT_MANIFEST, GLOB.cardinals, TRUE, user) // We can only set our decoy icon once. This proc is sort of expensive on generation.
+		clone_icon = get_flat_human_icon("[user.real_name] decoy", null, user.client?.prefs, DUMMY_HUMAN_SLOT_MANIFEST, GLOB.cardinals, TRUE, user) // We can only set our decoy icon once. This proc is sort of expensive on generation.
```

Comment:
using the prefs likely means it'll grab whichever slot is selected in char setup, including another char's